packages/edit-site/src/components/welcome-guide/template.js

```diff
@@ -10,29 +10,23 @@ import { store as editorStore } from '@wordpress/editor';
 /**
  * Internal dependencies
  */
-import { store as editSiteStore } from '../../store';
+import useEditedEntityRecord from '../use-edited-entity-record';
 
 export default function WelcomeGuideTemplate() {
 	const { toggle } = useDispatch( preferencesStore );
 
-	const isVisible = useSelect( ( select ) => {
-		const isTemplateActive = !! select( preferencesStore ).get(
-			'core/edit-site',
-			'welcomeGuideTemplate'
-		);
-		const isEditorActive = !! select( preferencesStore ).get(
-			'core/edit-site',
-			'welcomeGuide'
-		);
-		const { isPage } = select( editSiteStore );
-		const { getCurrentPostType } = select( editorStore );
-		return (
-			isTemplateActive &&
-			! isEditorActive &&
-			isPage() &&
-			getCurrentPostType() === 'wp_template'
-		);
+	const edited = useEditedEntityRecord();
+	const isPostTypeTemplate = edited.record.type === 'wp_template';
```

Comment:
```suggestion
	const isPostTypeTemplate = edited.record?.type === 'wp_template';
```

`edited.record` can be `false` on the first load.

Reply:
Aw, good catch. Yet in this case the optional chaining operator isn’t truly suited. Still, I’m not against trying to make this more explicit. I decided to leverage the `isLoaded` property in the return value 29bfbc06f8380cd3a729f55efe0b0add7539069f. Once that becomes `true` then `record` won’t be `false`.